### PR TITLE
Support writing pandas dataframes directly

### PR DESCRIPTION
**Pull request by changhiskhan**
addresses bug raised in #431

### CODE REVIEW

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -698,7 +698,9 @@ def write_dataset(
         The max number of rows before starting a new group (in the same file)
 
     """
-    if isinstance(data_obj, pa.Table):
+    if isinstance(data_obj, pd.DataFrame):
```
Comment (eddyxu) [inline python/python/lance/dataset.py:701]: Should we allow users to pass optional schema ?

Comment (changhiskhan) [reply python/python/lance/dataset.py:701]: i would say let's see if that's common enough of a need?
having the extra parameter is a little confusing if you're not passing in a DataFrame.
And the alternative is simple still: `lance.write_dataset(pa.Table.from_pandas(df, schema=schema), uri)`

wdyt?

Comment (eddyxu) [reply python/python/lance/dataset.py:701]: [`pyarrow.dataset.write_dataset()`](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.write_dataset.html) has schema as optional parameter, we could be consistent with pyarrow API for least surprise ?

**Review by eddyxu — APPROVED** [review]
LGTM

**Pull request merged**